R/compare.r

```diff
@@ -145,7 +145,7 @@ compare.numeric <- function(x, y, ..., max_diffs = 10) {
 
   # If they're not the same type or length, fallback to default method
   equal <- paste0(equal, collapse = "\n")
-  if (!is.integer(x) && !is.numeric(y)) return(comparison(FALSE, equal))
```

Comment:
Currently:

```
> testthat::expect_equal(1L, 0)
Error: 1L not equal to 0
1 - 0 == 1
```

I guess this case (and others) will be handled differently if comparing classes of x and y.